core/src/main/java/com/datastax/oss/driver/internal/core/metadata/InitialNodeListRefresh.java

```diff
@@ -90,30 +83,16 @@ public Result compute(
     }
 
     ImmutableList.Builder<Object> eventsBuilder = ImmutableList.builder();
-
     for (DefaultNode newNode : newNodes.values()) {
-      if (findIn(contactPoints, newNode.getEndPoint()) == null) {
-        eventsBuilder.add(NodeStateEvent.added(newNode));
-      }
+      eventsBuilder.add(NodeStateEvent.added(newNode));
     }
     for (DefaultNode contactPoint : contactPoints) {
-      if (findIn(newNodes.values(), contactPoint.getEndPoint()) == null) {
-        eventsBuilder.add(NodeStateEvent.removed(contactPoint));
-      }
+      eventsBuilder.add(NodeStateEvent.removed(contactPoint));
```

Comment:
Will this change force control connection to always reconnect on initial node refresh?

Reply:
Yes, it happens in com.datastax.oss.driver.internal.core.control.ControlConnection.SingleThreaded#onStateEvent .

Reply:
These docs are useful to grasp it high level https://docs.datastax.com/en/developer/java-driver/4.15/manual/developer/admin/index.html

Reply:
What if we check whether to reuse the contact point node by host ID? Then we can avoid that.

Reply:
Nope, contact point nodes come without host ID, they are built in `com.datastax.oss.driver.internal.core.session.DefaultSession.SingleThreaded#init` -> `com.datastax.oss.driver.internal.core.metadata.MetadataManager#addContactPoints`. We can try to do first match by `EndPoint` though.